test/e2e/kafka_binding_test.go

```diff
@@ -116,8 +142,11 @@ func TestKafkaBinding(t *testing.T) {
 	}
 	for name, tc := range tests {
 		tc := tc
-		t.Run(name, func(t *testing.T) {
-			testKafkaBinding(t, tc.messageKey, tc.messageHeaders, tc.messagePayload, tc.expectedData)
+		t.Run(name+"-v1alpha1", func(t *testing.T) {
```

Comment:
Would it be worth the effort to have some struct with the versions to test and iterate over those here? (probably only worth if more versions need to be maintained than the current two in the future, not sure about how this is expected to evolve)

Reply:
How about we create an issue to track this and do as a follow up?

Reply:
good suggestion @nlopezgi.
things here can be extracted into a helper function and then the test functions can call the helpers.
if you don't mind, let's do that afterwards or maybe even later when we have other versions.